jbpm-test-coverage/src/test/java/org/jbpm/test/functional/timer/TimerUpdateTest.java

```diff
@@ -192,6 +197,45 @@ public void beforeProcessStarted(ProcessStartedEvent event) {
 
         Assertions.assertThat(kieSession.getProcessInstance(id)).isNull();
     }
+    
+    @Test(timeout = 30000)
+    public void updateTimerSubprocessLongerDelayTest() {
+        CountDownProcessEventListener countDownListener = new CountDownProcessEventListener(TIMER_SUBPROCESS_NAME, 1);
+        //delay is set for 5s
+        setProcessScenario(TIMER_SUBPROCESS_FILE);
+
+        kieSession.addEventListener(countDownListener);
+        final List<Long> list = new ArrayList<Long>();
+        kieSession.addEventListener(new DefaultProcessEventListener() {
+            @Override
+            public void beforeProcessStarted(ProcessStartedEvent event) {
+                list.add(event.getProcessInstance().getId());
+            }
+            @Override
+            public void afterNodeLeft(ProcessNodeLeftEvent event) {
+                if (TIMER_SUBPROCESS_NAME.equals(event.getNodeInstance().getNodeName())) {
+                    System.setProperty(TIMER_FIRED_TEXT, "");
+                    System.setProperty(TIMER_FIRED_TIME_PROP, System.currentTimeMillis()+"");
```

Comment:
Everything looks great but I was thinking a little bit if there is something missing or not in "" string so I would rather prefer String.valueOf() or Long.toString() so there is no confusion. Otherwise than that it looks OK.

Reply:
fixed